File: MVVM_android_java/app/src/main/java/com/example/mvvm_android_java/MainActivity.java
package com.example.mvvm_android_java;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import android.os.Bundle;

import com.example.mvvm_android_java.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
    private final int nameDemo = 13;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMainBinding activityMainBinding = DataBindingUtil.setContentView(this, R.layout.activity_main);
            LoginViewModel loginViewModel = new LoginViewModel();
            activityMainBinding.setLoginViewModel(loginViewModel);
    }
}

File: login_MVVM_kotlin/app/src/main/java/com/example/login_mvvm_kotlin/MainActivity.kt
package com.example.login_mvvm_kotlin

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.example.login_mvvm_kotlin.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val activityMainBinding: ActivityMainBinding =
            DataBindingUtil.setContentView(this, R.layout.activity_main)
        val loginViewModel = LoginViewModel()
        activityMainBinding.loginViewModel = loginViewModel
    }
}